### DISCUSSION TITLE
Déblocage d'IP

### DISCUSSION THREAD

**Discussion by tvoirand** [bokub/conso-api#42] [Aide] [answered]
Bonjour,

Merci beaucoup pour ce projet! 

Je souhaite l'utiliser pour analyser ma consommation et ainsi mieux gérer mon empreinte carbone. En particulier, je souhaite récupérer régulièrement mes nouvelles données pour les sauvegarder en local, au moyen de requêtes envoyées depuis mon VPS.

Or il semble que l'IP de mon VPS soit bloquée. J'ai testé quelques requêtes à la main, qui ont retourné une erreur 403 "Sorry, you have been blocked". 

Sauf erreur de ma part, je n'ai pas abusé sur la quantité de requêtes, car j'ai reçu cette erreur dès la première requête envoyée depuis mon VPS, et j'avais auparavant testé depuis mon PC local exactement le même code qui a fonctionné avec succès. 

C'est pourquoi je me demande si le blocage de mon VPS pourrait être dû à une autre raison, et s'il est possible de le débloquer? 
Voici le début de l'IP en question:
IPv4: 51.68.xxx.xxx
IPv6: 2001:41d0:701:xxxx::xxxx

Je comprends que ton serveur reçois parfois une quantité abusive de requête, et que tu as dû en conséquence mettre en place un système de blocage d'IP. Je suis désolé que cet effort supplémentaire de gestion des abus soit nécessaire.

Merci d'avance pour ton retour et bonne journée!

**Comment by bokub** [comment]
Hello ! Quand tu dis "mon VPS", c'est quoi exactement ?

J'ai mis un blocage sur `2001:41d0:701:1100.*` le 16 février car mon API s'est faite spammer (première ligne de la capture ci-dessous)

[![image](https://github.com/bokub/conso-api/assets/17952318/f48b0a32-6b88-4be5-9684-0f5bbe8c7739)](https://github.com/bokub/conso-api/actions/runs/7930548315/job/21653048333)

**Reply by tvoirand** [reply]
Merci pour ta réponse express!

Par VPS j'entends Serveur Privé Virtuel, une petite machine virtuelle que je loue chez un hébergeur pour des projets perso.

A propos du ton spam du 16 février, effectivement je ne pense pas que ça venait de moi car j'ai découvert ton service seulement en mars, et j'ai créé mon compte Enedis le 12 mars.

Peut-être que le spam datant de février provenait d'une autre IP qui partage le même masque que la mienne? 

J'espère que c'est possible alors de débloquer mon adresse malgré tout. Si besoin je peux te partager plus de chiffres de mon IP.

**Comment by bokub** [comment, marked as answer]
OK j'ai retiré ton IP de la liste de blocage. J'espère que la personne qui partageait ton IP a arrêté de faire n'importe quoi, sinon tu seras bloqué à nouveau cette nuit malheureusement 😕 

Puis-je te te demander chez **qui** tu loue ton VPS ? (c'était ça la question que je voulais poser mais je me suis mal exprimé)

**Comment by tvoirand** [comment]
Ah oui j'avais mal compris pour le VPS au temps pour moi! Je suis chez OVH.

Merci beaucoup pour le déblocage. Je confirme que j'ai bien pu effectuer une requête aujourd'hui.

Je ne maîtrise pas trop ces sujets et ne sais pas par quel système tu effectue le blocage, mais si je comprends bien les requêtes abusives que tu avais identifiées provenaient d'une autre IP proche de la mienne. J'espère qu'il n'y en aura pas à nouveau. Si cela arrive, je peux aussi te partager mon IP complète, au cas où tu peux whitelister une IP précise.

**Reply by bokub** [reply]
J'utilise le WAF de Cloudflare pour bloquer les requêtes en fonction de l'IP.

Pour les IPv4 je bloque une seule IP, donc aucun problème, mais pour les IPv6 je suis obligé d'utiliser une notation CIDR avec 64 bits max donc uniquement les 4 premiers groupes de caractères (`aaaa:bbbb:cccc:dddd::/64`).

Si ton IP commence par les mêmes caractères, tu seras bloqué aussi.

Oui effectivement je peux aussi whitelist une IP précise, mais le nombre est très limité

**Reply by tvoirand** [reply]
D'accord, merci pour l'explication.

Il semble que mon voisin d'IP ait arrêté ses frasques, j'accède bien au service depuis plusieurs semaines sans problème.
